lib/core/src/restore.rs

```diff
@@ -0,0 +1,1164 @@
+            raw_tx_map
+                .into_iter()
+                .partition(|(_, tx)| tx.balance.values().sum::<i64>() < 0);
+
+        Self {
+            outgoing_tx_map,
+            incoming_tx_map,
+        }
+    }
+}
+
+trait PartialSwapState {
+    /// Determine partial swap state, based on recovered chain data.
+    /// For example, it cannot distinguish between [PaymentState::Created] and [PaymentState::TimedOut],
+    /// and in some cases, between [PaymentState::Created] and [PaymentState::Failed].
+    fn get_partial_state(&self) -> PaymentState;
+}
+
+pub(crate) struct RecoveredOnchainDataSend {
+    lockup_tx_id: Option<HistoryTxId>,
+    claim_tx_id: Option<HistoryTxId>,
+    refund_tx_id: Option<HistoryTxId>,
+}
+impl PartialSwapState for RecoveredOnchainDataSend {
+    fn get_partial_state(&self) -> PaymentState {
+        match &self.lockup_tx_id {
+            Some(_) => match &self.claim_tx_id {
+                Some(_) => PaymentState::Complete,
+                None => match &self.refund_tx_id {
+                    Some(refund_tx_id) => match refund_tx_id.confirmed {
+                        true => PaymentState::Failed,
+                        false => PaymentState::RefundPending,
+                    },
+                    None => PaymentState::Pending,
+                },
+            },
+            None => PaymentState::TimedOut,
+        }
+    }
+}
+
+pub(crate) struct RecoveredOnchainDataReceive {
+    lockup_tx_id: Option<HistoryTxId>,
+    claim_tx_id: Option<HistoryTxId>,
+}
+impl PartialSwapState for RecoveredOnchainDataReceive {
+    fn get_partial_state(&self) -> PaymentState {
+        match (&self.lockup_tx_id, &self.claim_tx_id) {
+            (Some(_), Some(claim_tx_id)) => match claim_tx_id.confirmed {
+                true => PaymentState::Complete,
+                false => PaymentState::Pending,
+            },
+            (Some(_), None) => PaymentState::Pending,
+            // TODO How to distinguish between Failed and Created (if in both cases, no lockup or claim tx present)
+            //   See https://docs.boltz.exchange/v/api/lifecycle#reverse-submarine-swaps
+            _ => PaymentState::Failed,
+        }
+    }
+}
+
+pub(crate) struct RecoveredOnchainDataChainSend {
+    lbtc_user_lockup_tx_id: Option<HistoryTxId>,
+    lbtc_refund_tx_id: Option<HistoryTxId>,
+    btc_server_lockup_tx_id: Option<HistoryTxId>,
+    btc_claim_tx_id: Option<HistoryTxId>,
+}
+impl PartialSwapState for RecoveredOnchainDataChainSend {
+    fn get_partial_state(&self) -> PaymentState {
+        // TODO How to detect TimedOut state?
+        //     TimedOut: This covers the case when the swap state is still Created and the swap fails to reach the
+        //     Pending state in time. The TimedOut state indicates the lockup tx should never be broadcast.
+        match &self.lbtc_user_lockup_tx_id {
+            Some(_) => match &self.btc_claim_tx_id {
+                Some(_) => PaymentState::Complete,
```

Comment:
Shouldn't we check if the btc_claim_tx is confirmed or not? If it is still not confirmed should the payment be still in Pending state?

Reply:
I followed the existing state handling logic, where we mark it as Complete as soon as we broadcast the claim:

https://github.com/breez/breez-sdk-liquid/blob/dc0e347e2b1604e8187f30eedc10486323522b91/lib/core/src/chain_swap.rs#L580-L603

Reply:
I see, indeed it is consistent. Question is do we have an issue with chain swap then? @dangeross shouldn't we make the chain swap completed only after our claim tx i s confirmed?

Reply:
It is currently marking the swap complete for both directions when the claim is broadcast. 

Perhaps for the incoming swap we can let the sync mark it complete. For the outgoing swap we are currently not monitoring the BTC address for confirmed transactions (but we could). What would you suggest @roeierez?

Reply:
@dangeross  For send I think we should especially because it is BTC mainnet where transactions may not confirm or go out of the mempool.
Also users see the swap as completed but still don't see the BTC on the destination address.
For receive, you are right the sync can do that same way we do for receive_swap.
But let's not decouple it with this PR. Once we fix it in a following PR we will fix here as well.

Reply:
The chain swap change was done in https://github.com/breez/breez-sdk-liquid/pull/450

The corresponding change to this PR was done in 97d62c5792b7b36da2e99f2926264116ccd59157